tests/service/SyntaxTreeTests/MemberTests.fs

```diff
@@ -362,3 +362,26 @@ type X =
         assertRange (4, 13) (4, 19) mInlineGet
         assertRange (5, 12) (5, 18) mInlineSet
     | ast -> Assert.Fail $"Could not get valid AST, got {ast}"
+
+[<Test>]
+let ``ImplicitCtor with as keyword`` () =
+    let parseResults = 
+        getParseResults
+            """
+type internal CompilerStateCache(readAllBytes: string -> byte[], projectOptions: FSharpProjectOptions)
+//#if !NO_TYPEPROVIDERS
+    as this =
+//#else
+//     =
+// #endif
```

Comment:
```suggestion
    as this =
```

Reply:
I prefer to keep this test as is. It better reflects my actual problem and will ensure it will not get broken over time.